src/Runtime/LoopIterator.php

```diff
@@ -45,11 +45,14 @@ public function __construct($seq)
             $this->seq = new \ArrayIterator($seq);
         } elseif ($seq instanceof \IteratorAggregate) {
             do {
+                /** @var \Iterator<TKey, TValue> $seq */
                 $seq = $seq->getIterator();
             } while ($seq instanceof \IteratorAggregate);
             $this->seq = $seq;
-        } elseif (is_iterable($seq)) {
```

Comment:
Is iterable is not always an Iterator but a Traversable.

So we technically have to use IteratorIterator if it's not an Iterator.

Reply:
Can you add a test that would break without this change?